app/views/posts/index.html.erb

```diff
@@ -17,7 +17,7 @@
       <%= link_to 'Create New Post', new_user_post_path(@user), class: 'btn btn-primary col-md-2' %>
     </div>
     <ul class='table table-responsive table-hover p-0'>
-      <% @user.recent_posts.each do |post| %>
+      <% @user.recent_posts.includes(:comments, :likes).each do |post| %>
         <li class='card mt-2 p-2'>
```

Comment:
- [ ] To solve the N+1 problem when fetching all posts and their comments for a user, Kindly you should use `@posts = @user.posts.includes(:comments)` instead of `@posts = @user.posts` in the `post#index` action inside your `posts_controller.rb`.

https://github.com/Eccb7/Pi-Blogs/blob/d229191aaf0ba4e5b8b46b70a43009439dd43dd3/app/controllers/posts_controller.rb#L4-L6